### PR TITLE
POC for AWS Url Specs

### PR DESCRIPTION
**Pull request by bikeath1337**
I am looking for the best pattern to repeat similar specs and decided to try a macro here. I also tried to pull out the specific spec set into another file, thus the update to package.json.

I still need to DRY up the common build/teardown logic I copied from evaporate.spec.js. Suggestions welcome.

I'm sort of following what I learned from Ruby RSpec...

### CODE REVIEW

**Comment by bikeath1337** [comment]
Hi, there, @jakubzitny . Was wondering what your timing was for reviewing this. I could have used the work in this branch to spec the fixes I am ready to release. Thanks!

**Comment by jakubzitny** [comment]
Hey, I'll have a look in few hours.

File: test/aws_url.spec.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  const config = Object.assign({}, baseAddConfig, {
+    complete: _handleUploadComplete.bind(obj),
+  })
+
+  evaporate.add(config)
+}
+
+// cloudfront, aws_url, awsRegion and s3Acceleration
+
+test.cb('should respect awsRegion and cloudfront when not defined', (t) => {
```
Comment (jakubzitny) [inline test/aws_url.spec.js:93]: Calling `test.cb` is not really needed for these cases. That's only when you want the test to be ended in a callback. Here you can just use `test(` and you don't need to pass no `t` or call `t.end()`.

File: test/aws_url.spec.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  global.setTimeout = (fc) => fc()
+})
+
+test.after(() => {
+  server.restore()
+})
+
+// test that AWS Url obeys the contract
+
+function testAwsUrl(obj, t, input, expected) {
```
Comment (jakubzitny) [inline test/aws_url.spec.js:74]: Passing `this`/`obj` all the time and binding it to the callback manually like this is not very cool.

Usually the stuff that's being tested in a test should be checked in `expect` inside that particular test, not in another reused function. That way when a test fails you check the line where it failed, you check out the code and you already see what's wrong.

@bikeath1337 let me try to refactor this a bit and submit it into the main repo.

File: test/aws_url.spec.js (hunk 1 above)
Comment (jakubzitny) [inline test/aws_url.spec.js:93]: Sorry, I just realized these actually DO end in callbacks. Ignore the previous comment.

File: test/aws_url.spec.js (hunk 2 above)
Comment (jakubzitny) [inline test/aws_url.spec.js:74]: Actually it's pretty good 😀.

I just removed useless imports from the file and I am sending the PR.

**Pull request closed without merging**